torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -627,7 +640,7 @@ void ProcessGroupNCCL::workCleanupLoop() {
     for (auto it = workList_.begin(); it != workList_.end();
          /* no increment*/) {
       auto& work = *it;
-      if (work->isCompleted()) {
+      if (work->isCompletedAndThrowException()) {
         // Handle Exceptions on failed GPU operations and remove completed
         // workNCCL objects from work vector.
         if (work->finishedGPUExecution()) {
```

Comment:
We probably need to remove this if we're already checking above?

Reply:
Right, this is removed in the latest version.